test/test_nn.py

```diff
@@ -7471,6 +7492,36 @@ def test_conv_cudnn_nhwc(self):
         self.assertEqual(conv.bias.grad, ref_conv.bias.grad)
         self.assertEqual(input.grad, ref_input.grad)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @unittest.skipIf(not TEST_CUDNN, "needs cudnn")
+    @skipIfRocm
+    def test_conv_cudnn_nhwc_size_1_filter(self):
```

Comment:
Same here, please move into device-generic class, and use new-style decorators to run only on cuda with cudnn